addons/mail/static/src/core/common/thread.js

```diff
@@ -173,6 +188,14 @@ export class Thread extends Component {
         return this.state.showJumpPresent ? PRESENT_THRESHOLD - 200 : PRESENT_THRESHOLD;
     }
 
+    get preferenceButtonText() {
+        const [, before, inside, after] =
+            _t(
+                "<button>Change your preferences</button> to receive new notifications in your Inbox."
+            ).match(/(.*)<button>(.*)<\/button>(.*)/) ?? [];
+        return { before, inside, after };
+    }
```

Comment:
I understand that we want to provide context for the translations and all. But this feel like an over engineered solution. the whole `match` is useless processing since you are already knowing witch string you need. The result will always be the same.
What will happen when the translator is going to remove the button tag? Or just make a mistake and remove `/` for he closing button? This will happen, for sure, and the translation will be broken.
This is a very fragile solution IMHO.

Also, why do you need to guard `?? []`? Match should always return the same array.


cc: @oomsveta

Reply:
The guard is a protection for the situation you mentioned. The translator can break the code (which is not always but can be a case) so the match can return nothing. In this way, the `[]` prevents the code from throwing errors. We definitely want to find a better way to handle this, but this is what we can have for now to handle the translation problem. In the future, @oomsveta will provide a better way to translate in the XML file but it needs time.

Reply:
So, in case the guard is triggered, nothing is displayed? And the translator has no feedback that he just broken something?
I think we should wait for the better solution instead of doing half baked solution.

Reply:
> I think we should wait for the better solution instead of doing half baked solution.

I think we should initiate work on having that solution now. I think this logic can be moved in an `l10n` utils function.
If we keep fleeing away from this issue with poor UX, e.g. by putting a link on the whole sentence or no link at all, then we'll always be weaker than competition and that's not good.